overlord/devicestate/firstboot.go

```diff
@@ -97,11 +97,11 @@ func populateStateFromSeedImpl(st *state.State, opts *populateStateFromSeedOptio
 	mode := "run"
 	sysLabel := ""
 	preseed := false
-	hasModeenv := false
+	isCoreBoot := false
```

Comment:
this should be:

isCoreBoot = !release.OnClassic

then the setting to true below is also correct

Reply:
I've tweaked this in a slightly different way that I think is more readable, let me know if you prefer what you suggest.

Reply:
it's ok, my thinking is that we might need isCoreBoot in other parts of the function later, but it's a hypothetical at this point, so it's fine either way, we can changed again later, thanks